File: data/derived_cot/rq1_traces/translation/qwen/java_to_py/code_output/ChandrasekharSieve.py
class ChandrasekharSieve:
    def __init__(self, n):
        self.n = n
        self.primes = self.generate_primes()

    def generate_primes(self):
        if self.n < 2:
            return []
        
        sieve = [True] * (self.n + 1)
        sieve[0] = sieve[1] = False
        
        p = 2
        while p * p <= self.n:
            if sieve[p]:
                start = p * p
                # Mark multiples starting from p*p
                for i in range(start, self.n + 1, p):
                    sieve[i] = False
            p += 1
        
        primes = [i for i, is_prime in enumerate(sieve) if is_prime]
        return primes

    def get_primes(self):
        return self.primes

if __name__ == '__main__':
    cs = ChandrasekharSieve(20)
    print(cs.get_primes())

File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/ChandrasekharSieve.cpp
#include <vector>
#include <iostream>

class ChandrasekharSieve {
private:
    int n;
    std::vector<int> primes;

    void generatePrimes() {
        if (n < 2) {
            return;
        }

        std::vector<bool> sieve(n+1, true);
        sieve[0] = false;
        sieve[1] = false;

        int p = 2;
        while (p * p <= n) {
            if (sieve[p]) {
                for (int i = p * p; i <= n; i += p) {
                    sieve[i] = false;
                }
            }
            p++;
        }

        for (int i = 2; i <= n; i++) {
            if (sieve[i]) {
                primes.push_back(i);
            }
        }
    }

public:
    ChandrasekharSieve(int n) : n(n) {
        generatePrimes();
    }

    std::vector<int> getPrimes() {
        return primes;
    }
};

int main() {
    ChandrasekharSieve cs(20);
    std::vector<int> primes = cs.getPrimes();
    std::cout << "[";
    for (size_t i = 0; i < primes.size(); i++) {
        if (i > 0) std::cout << ", ";
        std::cout << primes[i];
    }
    std::cout << "]" << std::endl;
    return 0;
}